packages/web/src/js/Collapse.ts

```diff
@@ -59,100 +62,133 @@ class Collapse extends BaseComponent {
+
+      if (!children) {
+        return;
+      }
+
+      for (const item of children) {
+        const itemTrigger = SelectorEngine.findOne(`[data-toggle="${NAME}"]`, item);
+        const instance = Collapse.getInstance(itemTrigger);
+
+        if (instance?.state?.open && itemTrigger !== trigger) {
+          instance?.hide();
+        }
+      }
+    }
   }
 
-  appendNodeToParentHandler() {
-    if (this.state.collapsed) {
+  async appendNodeToParentHandler() {
+    if (this.state.open) {
       const contentEl = this.target?.children[0];
       const innerHtml = contentEl ? contentEl?.innerHTML : this.target?.innerHTML;
       const originalHtml = this.parent?.innerHTML;
       if (this.parent) {
         this.parent.innerHTML = `${originalHtml}${innerHtml}`;
       }
-      setTimeout(() => this.target?.remove(), 100);
+
+      await this.target?.remove();
     }
   }
 
-  adjustCollapsibleElementHeightHandler(collapsed: boolean = this.state.collapsed) {
+  adjustCollapsibleChildrenHeightHandler(zeroHeight?: boolean) {
     if (this.target) {
       const content = this.target.children[0];
       const bounds = content.getBoundingClientRect();
-      this.target.style.height = collapsed ? `${bounds.height}px` : '0';
-      if (this.target) EventHandler.trigger(this.target, collapsed ? EVENT_SHOWN : EVENT_HIDDEN);
+      this.target.style.height = `${zeroHeight ? '0' : bounds.height}px`;
+    }
+  }
+
+  adjustCollapsibleElementHeightHandler(open: boolean = this.state.open) {
+    this.adjustCollapsibleChildrenHeightHandler(!open);
+    if (this.target) {
+      EventHandler.trigger(this.target, open ? EVENT_SHOWN : EVENT_HIDDEN);
     }
   }
 
-  updateTriggerElementHandler(collapsed: boolean = this.state.collapsed) {
-    const triggers = selectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
+  updateTriggerElementHandler(open: boolean = this.state.open) {
+    const triggers = SelectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
     const updateElement = (element: Element | HTMLElement) => {
       element.setAttribute(ARIA_CONTROLS_ATTRIBUTE, this.meta.id);
-      element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, String(collapsed));
-      element.classList.toggle(CLASSNAME_EXPANDED, collapsed);
-      if (this.meta.hideOnCollapse && collapsed) {
+      element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, String(open));
+      if (this.meta.hideOnCollapse && open) {
         element.remove();
         this.appendNodeToParentHandler();
         this.onDestroy();
       }
     };
-    this.state.collapsed = collapsed;
+    this.state.open = open;
     if (triggers.length === 1) {
       updateElement(this.element);
     } else {
-      triggers.forEach((trigger) => updateElement(trigger));
+      triggers.map((trigger) => updateElement(trigger));
     }
   }
 
-  updateCollapsibleElementHandler(collapsed: boolean = this.state.collapsed) {
+  updateCollapsibleElementHandler(open: boolean = this.state.open) {
```

Comment:
You are passing down the `open` property which you can use directly as `this.state.open`. Try to rethink this logic if it is really necessary to pass down this `open` flag and instead of it just using `this.state.open`.

Reply:
fixed already